Library/Formula/bullet.rb

```diff
@@ -44,11 +44,20 @@ def install
 
     args << "-DUSE_DOUBLE_PRECISION=ON" if build.with? "double-precision"
 
-    args << "-DBUILD_DEMOS=OFF" if build.without? "demo"
+    with_demo = build.with? "demo"
+    if with_demo && (build.with?("shared") || build.with?("framework"))
+      # Related to the following warnings when building --with-shared --with-demo
+      # https://gist.github.com/scpeters/6afc44f0cf916b11a226
+      opoo "Installed demos don't work with shared libraries"
+      puts <<-EOS.undent
+        Demos will be disabled due to the use of shared libraries.
+        Static libraries are required for demos installed by homebrew.
+      EOS
+      with_demo = false
+    end
+    args << "-DBUILD_BULLET2_DEMOS=OFF" unless with_demo
```

Comment:
Instead of silently correcting the behavior, Let's error out.

```ruby
if build.with?("demo") && (build.with?("shared") || build.with?("framework"))
  raise "Installed demos don't work with shared libraries"
end

args << "-DBUILD_BULLET2_DEMOS=OFF" unless build.with? "demo"
```